### PR TITLE
feat: balance check added in fund mov withdrawal

### PR DESCRIPTION
**Pull request by ZweHMyat**
- Participant settlement account balance checked before withdrawal fund movement approval

### CODE REVIEW

**Comment by elnyry-sam-k** [comment]
@Zwe333 

I was checking circieCI for this and the coverage report shows zero:


```
> @********/participant-bc@0.1.0 coverage:gen-report
> nyc report --reporter=lcov --reporter=text --temp-dir=./coverage

----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |                   
----------|---------|----------|---------|---------|-------------------
```

which is fine..

I want to confirm if this is accurate or not; to confirm if its a CI job / command issue or not.

Thank you!

File: packages/participants-svc/src/domain/participant_agg.ts [hunk 1]
```diff
@@ -1583,17 +1584,40 @@ export class ParticipantAggregate {
...
+        if (fundsMov.direction === ParticipantFundsMovementDirections.FUNDS_WITHDRAWAL) {
+            // Get the account from account and balance adapter
+            const updatedSettlementAcc = await this._accBal.getAccount(settlementAccount.id);
+            if (!updatedSettlementAcc) {
+                throw new AccountNotFoundError(
+                    `Could not get settlement account from accountsAndBalances adapter for participant id: ${participant.id}`
+                );
+            }
+            
+            if (!updatedSettlementAcc.balance) {
```
Comment (bushjames) [inline packages/participants-svc/src/domain/participant_agg.ts]: I think this is an unsafe way of checking for a balance. Note that js will return true in some circumstances you do not wish to proceed here e.g. if balance is a string, or a negative integer. I would suggest a little more rigour here in deciding if the balance property is "correct" at this point.

**Review by bushjames — CHANGES_REQUESTED** [review]

**Comment by ZweHMyat** [comment]
> @Zwe333
> 
> I was checking circieCI for this and the coverage report shows zero:
> 
> ```
> > @********/participant-bc@0.1.0 coverage:gen-report
> > nyc report --reporter=lcov --reporter=text --temp-dir=./coverage
> 
> ----------|---------|----------|---------|---------|-------------------
> File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
> ----------|---------|----------|---------|---------|-------------------
> All files |       0 |        0 |       0 |       0 |                   
> ----------|---------|----------|---------|---------|-------------------
> ```
> 
> which is fine..
> 
> I want to confirm if this is accurate or not; to confirm if its a CI job / command issue or not.
> 
> Thank you!

Hi @elnyry-sam-k the participant svc and other packages still don't have unit tests. Only implementation tests. That's why the coverage is zero.

File: packages/participants-svc/src/domain/participant_agg.ts (hunk 1 above)
Comment (ZweHMyat) [reply packages/participants-svc/src/domain/participant_agg.ts]: Got it and will take care of it. Thanks.

Comment (pedrosousabarreto) [reply packages/participants-svc/src/domain/participant_agg.ts]: This check is to conclude an account does not exist or has a bad balance value and break early, I would say this is being cautious.
Anyway, @Zwe333 please try to convert to number before processing with the comparison.

**Review by pedrosousabarreto — APPROVED** [review]
Looks good to me.

Thanks Zwe

**Pull request merged**